### ISSUE TITLE
Help interoperating with openssl command line interface

### ISSUE THREAD

**Issue by g105b** [ricmoo/aes-js#66]
I have created a cyphertext of a message using AES-128-CTR and have a copy of it in a hexadecimal string: `b02ff0b0874acaddf5149b1c9c`

The code I used to generate the above cyphertext is the example in the readme for "CTR - Counter (recommended)", with the `text` variable changed. Therefore the key is `[ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16 ]`.

I would like to learn how to decrypt this using openssl on the command line, but I can never work out how to get the original text back from the input hex string "b02ff0b0874acaddf5149b1c9c".

I do not understand how the IV is being set in the CTR example in the readme -- is it set when creating the `new aesjs.Counter(5)`, and if so how do I specify starting the counter from the arbitrary number 5 on the command line?

Here's what I've tried so far:

```
echo -n "b02ff0b0874acaddf5149b1c9c" | xxd -revert -plain | openssl aes-128-ctr -K "0102030405060708090A0B0C0D0E0F10" -iv 5 -d
```

Expected output: The plaintext input
Actual output: `Ðä¶ÎèCñ¹░¥8u`

Rationale behind my example above:

+ Echoing the cyphertext without newline added into xxd to convert it back into binary
+ Using the aes-128-ctr key as a hexidecimal string
+ As the counter is set to start from 5 for some reason, I've set the IV to 5 as that seems the only way to match this arbitrary choice up

Any help in pointing me in the right direction will be appreciated.

**Comment by g105b** [comment]
Thanks @RokerHRO I'll give it another try with hex values and report back here.

**Comment by Ruffio** [comment]
@g105b any news? Can this issue be closed?

**Comment by g105b** [comment]
@Ruffio thanks for the poke... I have just tried collating all my thoughts to check if I can get this to work. Sadly no, but I have documented by attempts - hopefully it is something _I_ am doing wrong.

You can see my test page here: https://github.com/g105b/aesjs-test/blob/master/index.php

Type `Hello` into the box, press "Encrypt with aes-js" and the text box will change to hex value `108790a295`. If you press "Decrypt with aes-js" you'll see the message appear: `Decrypted message from AES-JS: Hello` ... all good!

Now press "Decrypt with PHP" and you'll see the message: `Decrypted message from PHP: KNz�` which is obviously suggesting that the decryption process is missing a vital step, but I have tried everything I know.

Basically, encrypting/decrypting between PHP scripts is easy. Encrypting/decrypting between JS scripts using this library is easy. Encrypting/decrypting using openssl on the terminal is easy. But trying to get the same formats working between JS and PHP is seeming really difficult!

Any advice on my PHP decryption will be greatly appreciated!

**Comment by g105b** [comment]
Something I don't understand that might be causing my lack of success on this is how IVs are handled with the counter method of AES encryption.

Using this library, `var aesCtr = new aesjs.ModeOfOperation.ctr(key).encrypt("message")` seems to encrypt fine. From the docs I infer that without specifying, the counter starts from zero. However, when using counter mode, I see no reference to IV. Equally, with openssl I see no reference to the counter start position - maybe I am missing something vital here?

**Comment by MWalid** [comment]
I was able to encrypt with this library and decrypt with Java, @g105b please check this post, it may help you, https://stackoverflow.com/q/52304774/2407522

**Comment by MWalid** [comment]
@g105b got it to work with php as well, please have a look at https://github.com/MWalid/aesjs-test/blob/master/index.php

**Comment by g105b** [comment]
@MWalid Thank you so much for helping me understand this! Your simple PHP code is so clear as to how the IV is used in the CTR mode, and it works a treat with no problems.

I had other questions too regarding IV and hex, but it all makes sense now, just took you to show me a working copy in a language I was familiar with.

Thank you again!

**Comment by MWalid** [comment]
Hi @g105b, my pleasure ...
